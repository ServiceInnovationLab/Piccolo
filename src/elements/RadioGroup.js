import React from 'react';
import styled from 'styled-components';

const RadioGroupWrapper = styled.div`
  display: flex;
  flex-flow: column;
`;

const RadioGroupInner = styled.div`
  flex: 1 0 auto;
  display: flex;
  justify-content: center;
  flex-flow: column;
  position: relative;
  @media(min-width: 1024px) {
    margin-top: .5em;
  }
`;

const Label = styled.label`
  display: inline-block;
  cursor: pointer;
  line-height: 1.5;
  margin: 0;
  font-size: 18px;
  min-width: 70px;
  :first-child > span {
    border-top-left-radius: 25px;
    border-bottom-left-radius: 25px;
    border-right: 4px solid #2f3045;
  }
  :last-child > span {
    border-top-right-radius: 25px;
    border-bottom-right-radius: 25px;
    border-left: 4px solid #2f3045;
  }
  + label {
    margin-left: 24px;
  }
`;

const Radio = styled.input.attrs({ type: 'radio' })`
  position: absolute;
  left: -9999px;
  margin: 0;
  display: inline-block;
  border: 2px solid transparent;
  border-radius: 10px;
  padding: 10px 20px;
  :checked + span {
    background-color: #019dff;
    border-color: #2f3045;
    color: #fff;
    font-weight: bold;
  }
`;

const Span = styled.span`
  display: inline-block;
  border: 4px solid #2f3045;
  width: 100%;
  text-align: center;
  font-size: 20px;
  line-height: 1.5;
  color: #000;
  background: #fff;
  font-weight: 600;
  padding: 9.5px 10px;
  text-transform: capitalize;
`;

const RadioGroup = props => (
  <RadioGroupWrapper>
    <RadioGroupInner>
      <div>
        {props.values.map((val, i) => <Label key={i}>
          <Radio
              name={props.name}
              value={val}
              onClick={e => props.handleClick(e, props.name)}
          />
          <Span>{val}</Span>
        </Label>)}
      </div>
    </RadioGroupInner>
  </RadioGroupWrapper>
);

export default RadioGroup;
