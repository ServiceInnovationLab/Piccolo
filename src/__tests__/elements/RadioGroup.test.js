import React from 'react';
import renderer from 'react-test-renderer';
import RadioGroup from '../../elements/RadioGroup';
import 'jest-styled-components';

test('RadioGroup renders correctly', () => {
  const tree = renderer.create(<RadioGroup values={['yes', 'no']} />).toJSON();
  expect(tree).toMatchSnapshot();
});
