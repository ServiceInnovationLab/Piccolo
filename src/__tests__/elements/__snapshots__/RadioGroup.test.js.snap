// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
}

.c1 {
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  position: relative;
}

.c2 {
  display: inline-block;
  cursor: pointer;
  line-height: 1.5;
  margin: 0;
  font-size: 18px;
  min-width: 70px;
}

.c2:first-child > span {
  border-top-left-radius: 25px;
  border-bottom-left-radius: 25px;
  border-right: 4px solid #2f3045;
}

.c2:last-child > span {
  border-top-right-radius: 25px;
  border-bottom-right-radius: 25px;
  border-left: 4px solid #2f3045;
}

.c2 + label {
  margin-left: 24px;
}

.c3 {
  position: absolute;
  left: -9999px;
  margin: 0;
  display: inline-block;
  border: 2px solid transparent;
  border-radius: 10px;
  padding: 10px 20px;
}

.c3:checked + span {
  background-color: #019dff;
  border-color: #2f3045;
  color: #fff;
  font-weight: bold;
}

.c4 {
  display: inline-block;
  border: 4px solid #2f3045;
  width: 100%;
  text-align: center;
  font-size: 20px;
  line-height: 1.5;
  color: #000;
  background: #fff;
  font-weight: 600;
  padding: 9.5px 10px;
  text-transform: capitalize;
}

@media (min-width:1024px) {
  .c1 {
    margin-top: .5em;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div>
      <label
        className="c2"
      >
        <input
          className="c3"
          type="radio"
          value="yes"
        />
        <span
          className="c4"
        >
          yes
        </span>
      </label>
      <label
        className="c2"
      >
        <input
          className="c3"
          type="radio"
          value="no"
        />
        <span
          className="c4"
        >
          no
        </span>
      </label>
    </div>
  </div>
</div>
`;
